src/Surfnet/AzureMfa/Domain/UserId.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types=1);
+
+/**
+ * Copyright 2019 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\AzureMfa\Domain;
+
+use Exception;
+use Surfnet\AzureMfa\Domain\Exception\InvalidUserIdException;
+
+class UserId
+{
+    const SEPARATOR = '|';
+    const VALID_UNIQUE_ID = '/^[a-z0-9]{6}-[a-z0-9]{4}$/';
+
+    /**
+     * @var string
+     */
+    private $userId;
+    /**
+     * @var EmailAddress
+     */
+    private $emailAddress;
+
+    public function __construct(string $userId)
+    {
+        if (empty($userId)) {
+            throw new InvalidUserIdException('An empty id was specified');
+        }
+
+        $pos = strpos($userId, self::SEPARATOR);
+        if ($pos === false) {
+            throw new InvalidUserIdException('An invalid id was specified');
+        }
+        $emailAddress = substr($userId, $pos + 1);
+        $uniquePrefix = substr($userId, 0, $pos);
+
+        $match = preg_match(self::VALID_UNIQUE_ID, $uniquePrefix);
+        if ($match !== 1) {
+            throw new InvalidUserIdException('An invalid id was specified');
+        }
+
+        try {
+            $this->emailAddress = new EmailAddress($emailAddress);
+        } catch (Exception $exception) {
+            throw new InvalidUserIdException('An invalid id was specified');
+        }
+
+        $this->userId = $userId;
+    }
+
+    public static function generate(EmailAddress $emailAddress): UserId
+    {
+        $length = 4;
+        // TODO: copied from Tiqr, is this safe to use?
```

Comment:
Can the todo be removed? In my opinion this is safe to use, but I see no clear benefit to separating the time based and the random part with a `-`. If we are going to use a separator, then I'd like to see only one being used in the string. a `-` might not be the best choice then, as this character can be in an email address.

Reply:
Removed the todo. As I said earlier I didn't want to invent the wheel and the Tiqr id works so I currently don't see the need to change it.